server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSessionServlet.scala

```diff
@@ -195,8 +207,17 @@ class InteractiveSessionServlet(livyConf: LivyConf)
 
   jpost[AddResource]("/:id/add-jar") { req =>
     withSession { lsession =>
-      val uri = new URI(req.uri)
-      lsession.addJar(uri)
+      addJarOrPyFile(req, lsession)
+    }
+  }
+
+  jpost[AddResource]("/:id/add-pyfile") { req =>
+    withSession { lsession =>
+      lsession.kind match {
+        case Spark() => throw new UnsupportedOperationException("End point unsupported for " +
```

Comment:
This will result in a 500 error which is not very helpful, should be `BadRequest`.

Also, the message is wrong: this is not a pyspark session. Should say "only supported for pyspark sessions". And end with a period.